packages/migrate/src/plugins/diagnostics-autofix.plugin.ts

```diff
@@ -65,24 +66,30 @@ export default class DiagnosticAutofixPlugin extends Plugin {
   /**
    * Prepares a hint message for engineer based on original diagnostic message and `this.hint`
    */
-  prepareHint(message: string | ts.DiagnosticMessageChain, hint?: string): string {
+  prepareHint(
+    message: string | ts.DiagnosticMessageChain,
+    diagnostic: ts.DiagnosticWithLocation,
+    getHint?: (
+      replacements: { [key: string]: string },
+      tsDiagnostic: ts.DiagnosticWithLocation
+    ) => string | undefined
+  ): string {
     message = ts.flattenDiagnosticMessageText(message, '. ');
-
-    if (hint) {
-      // Prepare a replacement dictionary
-      // e.g. {'{0}': 'p', '{1}': 'any'} for message "Parameter 'p' implicitly has an 'any' type."
-      const replacements: { [key: string]: string } =
-        message
-          // return ["'p'", "'any'"]
-          .match(/'[^']+'/gm)
-          // converts ["'p'", "'any'"] to {'{0}': 'p', '{1}': 'any'}
-          ?.reduce((a, v, i) => ({ ...a, [`{${i}}`]: v.replace(/^\W+|\W+$/g, '') }), {}) || {};
-
-      // Replaces {0}, {1}, ... placeholders with corresponding values from the original message
-      message = hint.replace(/{\d+}/gm, (key) => replacements[key] || key);
+    let hint: string | undefined;
+    // Prepare a replacement dictionary
+    // e.g. {'{0}': 'p', '{1}': 'any'} for message "Parameter 'p' implicitly has an 'any' type."
+    const replacements: { [key: string]: string } =
+      message
+        // return ["'p'", "'any'"]
+        .match(/'[^']+'/gm)
+        // converts ["'p'", "'any'"] to {'{0}': 'p', '{1}': 'any'}
+        ?.reduce((a, v, i) => ({ ...a, [`{${i}}`]: v.replace(/^\W+|\W+$/g, '') }), {}) || {};
+
+    if (getHint) {
+      hint = getHint(replacements, diagnostic);
```

Comment:
The benefit of original code is to have simple transforms, since all replacement logic is happening outside.
With getHint, engineer have "repeat" this logic in every function :(